### ISSUE TITLE
Debug testing errors/add more tests for loading rooms into game-state structs

### ISSUE THREAD

**Issue by mewthew** [uchicago-cs/chiventure#267]
After game-state's approved PR and fixed compiler issues in #261, our tests for loading rooms found in wdl/tests/test_room.c did not pass. We need to debug these issues and fix our tests such that everything compiles and rooms are correctly loaded into game-state structs in preparation for a PR to master for loading libobj structs into game-state structs.

**Comment by mewthew** [comment]
I have debugged the issues for loading rooms into game-state structs, and now, all of our previous tests work for loading rooms into games. For efficiency, I have created helper functions within test_room.c that checks room descriptions and connection IDs from game-state structs. These functions helped me create more tests that cover each room short description, long description, and connection ID within the YAML file located in examples/sample_yaml/connected_rooms.yaml. The implementation for the above can be found [here](https://github.com/uchicago-cs/chiventure/blob/wdl/fixing-compiler-errors/src/wdl/tests/test_room.c).
